core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -497,11 +497,18 @@ private[spark] class BlockManagerInfo(
 
     updateLastSeenMs()
 
+    var blockExists = false
```

Comment:
Although you could just assign `val blockExists = _blocks.containsKey(blockId)` once and reuse it instead of a `var`, it's minor. This looks good.